################################################

# Campylobacter Metagenomics Pipeline

################################################

### PATH to Metagenomes of Cases and Controls (Campylobacter)
/mnt/research/manninglab/Sequencing_Data/misc_sequencing/20161123_B_DNASeq_PE/ER*_NNNNNNNN-NNNNNNNN_L00*_R*.fastq.gz
/mnt/research/manninglab/Sequencing_Data/misc_sequencing/20150123_DNASeq_PE_metagenomes/ER*_NNNNNNNN-NNNNNNNN_L00*_R*.fastq.gz
/mnt/research/manninglab/Sequencing_Data/misc_sequencing/20170206_B_DNASeq_PE/ER*_NNNNNNNN-NNNNNNNN_L00*_R*.fastq.gz

### Merge paired-end reads
sample_names=(ER0* .. ER1*)

for sample in ${sample_names[@]}
do
cat ./${sample}_*_L001_R1_001.fastq.gz ${sample}_*_L002_R1_001.fastq.gz > /mnt/scratch/hansenzo/${sample}_Forward.fastq.gz; 
cat ./${sample}_*_L001_R2_001.fastq.gz ${sample}_*_L002_R2_001.fastq.gz > /mnt/scratch/hansenzo/${sample}_Reverse.fastq.gz
done

### From this point, the sample names "ERIN_Forward.fastq.gz" and "ERIN_Reverse.fastq.gz" will represent the files of interest

### Quality Assessment using FastQC
module load FastQC/0.11.7-Java-1.8.0_162

fastqc ERIN_Forward.fastq.gz;
fastqc ERIN_Reverse.fastq.gz

### Use Trimmomatic to perform trimming and adpater removal
module load Trimmomatic/0.38-Java-1.8.0_162
module load GCC/6.4.0-2.28 OpenMPI/2.1.2

java -jar $TRIM/trimmomatic PE -threads 28 -phred33 ERIN_Forward.fastq.gz ERIN_Reverse.fastq.gz trim.f.pe.fq.gz trim.f.se.fq.gz trim.r.pe.fq.gz trim.r.se.fq.gz ILLUMINACLIP:~/Database/illuminaClipping.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36;

# Note: the 'illuminaClipping.fa' files was obtained upon retrieval of sequencing data

### Now, the trimmed files are represented as 'ERIN_trim.f.pe.fq.gz' and 'ERIN_trim.r.pe.fq.gz'

### Re-run Quality Assessment with trimmed reads
fastqc ERIN_trim.f.pe.fq.gz;
fastqc ERIN_trim.r.pe.fq.gz

# Note: terminology of 'ERIN_trim.f/r.pe.fq.gz' is as follows:
      # trim = trimmed reads
      # f/r = forward or reverse
      # pe = paired end

### Removal of reads aligning to the human genome
module load icc/2018.1.163-GCC-6.4.0-2.28 impi/2018.1.163
module load Bowtie2/2.3.4.1

bowtie2 -p 5 -x ~/Database/HumanRef/human_DB -1 ERIN_trim.f.pe.fq.gz -2 ERIN_trim.r.pe.fq.gz -S ERIN_host_mapped.sam --threads 24

# Note: 'human_DB' refers to human RefSeq genomes (GRCh38 downloaded November 2018) available from the National Center for Biotechnology information (NCBI)

### Use SAMtools and BEDtools to clean up re-arrange the reads
module load GCC/6.4.0-2.28 OpenMPI/2.1.1
module load SAMtools
module load BEDTools 

samtools view -bS -@ 24 host_mapped.sam > host_mapped.bam
samtools view -b -@ 24 -f 12 -F 256 host_mapped.bam > host_removed.bam
samtools sort -n -@ 24 host_removed.bam > host_removed_sorted.bam
bedtools bamtofastq -i host_removed_sorted.bam -fq ERIN_trim.f.pe.hre.fq -fq2 ERIN_trim.r.pe.hre.fq
gzip ERIN_trim.f.pe.hre.fq
gzip ERIN_trim.r.pe.hre.fq

# Note: terminology of 'ERIN_trim.f/r.pe.hre.fq.gz' is as follows:
      # trim = trimmed reads
      # f/r = forward or reverse
      # pe = paired end
      # hre = Human genome REmoved
      
### Alignment to MEGARes, antibiotic resistance gene database

# MEGARes Version 1.0.1 was used for the analysis in this publication
# MEGARes files can be downloaded here: https://megares.meglab.org/download/index.php

module load icc/2018.1.163-GCC-6.4.0-2.28  impi/2018.1.163
module load ifort/2018.1.163-GCC-6.4.0-2.28  impi/2018.1.163
module load Bowtie2/2.3.4.1

bowtie2 -p 5 -x ~/Database/ARGs/MEGARes/MEGARes -1 ~/ERIN_trim.f.pe.hre.fq.gz -2 ~/ERIN_trim.r.pe.hre.fq.gz -S ./megares_mapped.sam


### Identify and characterize resistance genes using AmrPlusPlus tools

# These include resistomeanalyzer, rarefactionanalyzer, and SNPfinder
    # Information on resistomeanalyzer can be found here: https://github.com/cdeanj/resistomeanalyzer
    # Information on rarefactionanalyzer can be found here: https://github.com/cdeanj/rarefactionanalyzer
    # Information on SNPfinder can be found here: https://github.com/cdeanj/snpfinder

module load GNU/4.9.2

~/Documents/resistomeanalyzer/resistome -ref_fp ~/Database/ARGs/megares_database_v1.01.fasta -sam_fp megares_mapped.sam -annot_fp ~/Database/ARGs/megares_annotations_v1.01.csv -gene_fp ./gene.tsv -group_fp ./group.tsv -class_fp ./class.tsv -mech_fp ./mech.tsv -t 80; 

~/Documents/RarefactionAnalyzer/rarefaction -ref_fp ~/Database/ARGs/megares_database_v1.01.fasta -sam_fp ./megares_mapped.sam -annot_fp ~/Database/ARGs/megares_annotations_v1.01.csv -gene_fp ./rar_gene.tsv -group_fp ./rar_group.tsv -class_fp ./rar_class.tsv -mech_fp ./rar_mech.tsv -min 5 -max 100 -skip 5 -samples 1 -t 80;

~/Documents/snpfinder/snpfinder -amr_fp ~/Database/ARGs/megares_database_v1.01.fasta -sampe megares_mapped.sam -out_fp ~/snps.txt;


### The outputs generated by resistomeanalzyer, rarefactionanalzyer, and SNPfinder can be used for downstream analyses such as exploration of abundance, composition, and diversity.









